trin-core/src/types/header.rs

```diff
@@ -28,44 +29,76 @@ pub struct Header {
     /// Block receipts root.
     pub receipts_root: H256,
     /// Block bloom filter.
-    pub log_bloom: Bloom,
+    pub logs_bloom: Bloom,
     /// Block difficulty.
     pub difficulty: U256,
     /// Block number.
+    #[serde(deserialize_with = "de_hex_to_u64")]
     pub number: u64,
     /// Block gas limit.
     pub gas_limit: U256,
     /// Block gas used.
     pub gas_used: U256,
     /// Block timestamp.
+    #[serde(deserialize_with = "de_hex_to_u64")]
     pub timestamp: u64,
     /// Block extra data.
-    #[serde(serialize_with = "as_hex")]
+    #[serde(serialize_with = "se_hex")]
+    #[serde(deserialize_with = "de_hex_to_vec_u8")]
     pub extra_data: Vec<u8>,
     /// Block PoW mix hash.
     pub mix_hash: Option<H256>,
     /// Block PoW nonce.
-    #[serde(serialize_with = "raw_bytes")]
+    #[serde(deserialize_with = "de_hex_to_option_bytes")]
+    #[serde(serialize_with = "se_raw_bytes")]
     pub nonce: Option<Bytes>,
```

Comment:
Could we switch this type to `ethereum_types::H64` and get rid of the helper methods? I'm not sure if `Option` makes sense here, is it possible to have a header without nonce?

Reply:
Off the top of my head, I'm not sure. But we perform a check on this field on lines `117`, and `137`. So I think it's safe to assume that it is possible and we shouldn't change the core field type.

Reply:
`Option<H64>` seems to be fine, gets rid of the helper methods, and mirrors what is done [here](https://github.com/vorot93/ethereum-jsonrpc/blob/master/src/types/block.rs#L102)